WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductFromImage/AddProductFromImageViewModel.swift

```diff
@@ -25,8 +37,13 @@ final class AddProductFromImageViewModel: ObservableObject {
 
     // MARK: - Scanned Texts
 
+    @Published var scannedTexts: [ScannedTextViewModel] = []
     @Published private(set) var isGeneratingDetails: Bool = false
 
+    private var selectedScannedTexts: [String] {
+        scannedTexts.filter { $0.isSelected }.map { $0.text }
+    }
```

Comment:
💡 I'd suggest filtering text that is not empty and disabling the Regenerate button if this array is empty. I tried clearing all the texts and regenerating and get random unrelated titles and descriptions for the product 😅 

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/0d97e8f2-d9b0-492c-b478-a21c4e1c9774" width=320 />

Reply:
We can also toggle the selection for a text if it's empty.

Reply:
Good catch on this edge case. I was struggling with an observable property on the `scannedTexts` Published var, since each row's states are in a nested ObservableObject and it doesn't trigger updates to the parent view model when the nested object changes states. I'm leaving it a separate subtask in https://github.com/woocommerce/woocommerce-ios/issues/10180 when I can refactor the view model to unit test. In the meantime, I updated to only include non-empty selected texts in https://github.com/woocommerce/woocommerce-ios/pull/10220/commits/5daf6718b31b1f264426f428e3d6a31522376373